static/js/components/ranking_unit.tsx

```diff
@@ -132,6 +133,120 @@ export function getPointsList(
   return pointsList;
 }
 
+function RankingRow(props: {
+  point: RankingPoint;
+  highlightedDcid?: string;
+  entityType?: string;
+  apiRoot?: string;
+  onHoverToggled?: (placeDcid: string, hover: boolean) => void;
+  hideValue?: boolean;
+  scaling?: number[];
+  unit?: string[];
+  svNames?: string[];
+}): JSX.Element {
+  const {
+    point,
+    highlightedDcid,
+    entityType,
+    apiRoot,
+    onHoverToggled,
+    hideValue,
+    scaling,
+    unit,
+  } = props;
+  const [isVisible, setIsVisible] = React.useState(false);
+  const rowRef = React.useRef<HTMLTableRowElement>(null);
+  const urlFunc = React.useContext(RankingUnitUrlFuncContext);
+
+  React.useEffect(() => {
```

Comment:
is it possible to reuse this [lazyLoad](https://github.com/datacommonsorg/website/blob/92dd88ec5f3923311239c5afbb000a79daed44ab/static/js/shared/hooks.ts#L24) code w/ a margin of >= 200px to reduce the amount of onscreen load time?

Alternatively using the [useInView](https://www.npmjs.com/package/react-intersection-observer) hook, [example here](https://github.com/datacommonsorg/website/blob/92dd88ec5f3923311239c5afbb000a79daed44ab/static/js/apps/explore/result_header_section_legacy.tsx#L51C30-L51C39)

also, is there a way batch fetch the names instead of loading individually? That might be a bigger refactor.